src/Identicon/Generator/BaseGenerator.php

```diff
@@ -238,4 +179,63 @@ public function getPixelRatio()
     {
         return $this->pixelRatio;
     }
+
+    /**
+     * Convert the hash into an multidimensionnal array of boolean
+     *
+     * @return this
+     */
+    private function convertHashToArrayOfBoolean()
+    {
+        preg_match_all('/(\w)(\w)/', $this->hash, $chars);
+        foreach ($chars[1] as $i => $char) {
+            if ($i % 3 == 0) {
+                $this->arrayOfSquare[$i/3][0] = $this->convertHexaToBoolean($char);
+                $this->arrayOfSquare[$i/3][4] = $this->convertHexaToBoolean($char);
+            } elseif ($i % 3 == 1) {
+                $this->arrayOfSquare[$i/3][1] = $this->convertHexaToBoolean($char);
+                $this->arrayOfSquare[$i/3][3] = $this->convertHexaToBoolean($char);
+            } else {
+                $this->arrayOfSquare[$i/3][2] = $this->convertHexaToBoolean($char);
+            }
+            ksort($this->arrayOfSquare[$i/3]);
+        }
+
+        $this->color[0] = hexdec(array_pop($chars[1]))*16;
+        $this->color[1] = hexdec(array_pop($chars[1]))*16;
+        $this->color[2] = hexdec(array_pop($chars[1]))*16;
+
+        return $this;
+    }
+
+    /**
+     * Convert an heaxecimal number into a boolean
+     *
+     * @param string $hexa
+     *
+     * @return boolean
+     */
+    private function convertHexaToBoolean($hexa)
+    {
+        return (bool) intval(round(hexdec($hexa)/10));
+    }
+
+    private function convertColor($color)
+    {
+        $convertedColor = array();
+        if (is_array($color)) {
+            $convertedColor[0] = $color[0];
+            $convertedColor[1] = $color[1];
+            $convertedColor[2] = $color[2];
+        } else {
+            if (false !== strpos($color, '#')) {
+                $color = substr($color, 1);
+            }
+            $convertedColor[0] = hexdec(substr($color, 0, 2));
+            $convertedColor[1] = hexdec(substr($color, 2, 2));
+            $convertedColor[2] = hexdec(substr($color, 4, 2));
+        }
+
+        return $convertedColor;
+    }
```

Comment:
Normally I just fix the identation, but perhaps (I use auto fix of phpstorm) some other change have been made. Sorry for that. I can go back and make a rebase to assemble each commit to one, to have a more clear vision of all change.